### PR TITLE
Type definitions and tests, ES module build

### PR DESCRIPTION
**Pull request by johanventer**
## Description of the change

> Adds type definitions to the build.

> Based on Jayden's work in #9 but adds tsd for testing that the exported type definitions work.

> Also adds an ES module and CJS build for use in npm, with no bundled dependencies to reduce bundle sizes.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: DEV-3019
 - Process.st launch checklist: https://app.process.st/checklists/DEV-3019-squatch-js-23-s1dh-kN6ywgx6zQSSSlMAA

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [2.3.0] - 2021-05-19
+### Added
+- Typescript definitions
+- CommonJS/ES modules build for smaller bundle sizes when using squatch-js from NPM
+
```
Comment (loganvolkers) [inline CHANGELOG.md:13]:
```suggestion
### Changed
 - `main` is now a CommonJS build instead of a UMD build
```

**Review by loganvolkers — COMMENTED** [review]
Looks good minus question on changelog

File: src/utils/validate.ts [hunk 2]
```diff
@@ -58,5 +58,6 @@ export function validateConfig(raw: unknown): Required<ConfigOptions> {
 export function validateWidgetConfig(raw: unknown): WidgetConfig {
   if (!isObject(raw)) throw new Error("Widget properties must be an object");
   if (!assertProp(raw, "user")) throw new Error("Required properties missing.");
-  return raw;
+  // TODO: This should be better type checked
```
Comment (00salmon) [inline src/utils/validate.ts:61]: just want to make sure this wasn't a TODO we wanted to resolve now (not sure when we'll get back to squatchJS to address this TODO)

File: src/api/EventsApi.ts [hunk 3]
```diff
@@ -80,7 +79,8 @@ function _validateEvent(raw: unknown): UserEventInput {
     throw new Error("Fields required");
   if (!Array.isArray(raw.events))
     throw new Error("'events' should be an array");
-  return raw;
+  // TODO: Better type checking
```
Comment (00salmon) [inline src/api/EventsApi.ts:82]: same todo comment here

**Review by 00salmon — APPROVED** [review]
looks good to me, having types in there is awesome.

File: src/utils/validate.ts (hunk 2 above)
Comment (johanventer) [reply src/utils/validate.ts:61]: I want to implement something like https://github.com/woutervh-/typescript-is at some point. Compile time type guard functions for arbitrary data!

**Review by 00salmon — APPROVED** [review]

**Pull request merged**